patches/minecraft/net/minecraft/client/audio/SoundHandler.java.patch

```diff
@@ -0,0 +1,33 @@
+--- ../src-base/minecraft/net/minecraft/client/audio/SoundHandler.java
++++ ../src-work/minecraft/net/minecraft/client/audio/SoundHandler.java
+@@ -67,6 +67,7 @@
+     {
+         this.field_147697_e.func_148763_c();
+ 
++        Map<ResourceLocation, SoundList> resources = new java.util.IdentityHashMap<>();
+         for (String s : p_110549_1_.func_135055_a())
+         {
+             try
+@@ -79,6 +80,8 @@
+ 
+                         for (Entry<String, SoundList> entry : map.entrySet())
+                         {
++                            resources.put(new ResourceLocation(s, entry.getKey()), entry.getValue());
++                            if (false)
+                             this.func_147693_a(new ResourceLocation(s, entry.getKey()), entry.getValue());
+                         }
+                     }
+@@ -94,6 +97,13 @@
+             }
+         }
+ 
++        net.minecraftforge.fml.common.ProgressManager.ProgressBar resourcesBar = net.minecraftforge.fml.common.ProgressManager.push("Loading sounds", resources.size());
```

Comment:
You should move this code above the big for loop so that you don't need to keep track of an extra map.

Reply:
The map is needed for the max count passed to the `ProgressBar` constructor.

Reply:
Oh. Then this part is fine.
If you copy the data over, I suggest you using two `ArrayList`s or two `ArrayDeque`s so that you don't need to iterate over a hash map slowly when you keep track over registry entries and sound lists.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#size() ?

Reply:
Yep, that's how I'm using it.